src/libraries/System.Security.Cryptography/tests/X509Certificates/FindTests.cs

```diff
@@ -958,6 +958,233 @@ public static void TestFindByKeyUsage_NoMatch(object matchCriteria)
+                Assert.Equal(found, cert);
+                found = Assert.Single(coll.FindByThumbprint(hashAlgorithm, thumbprintHexLower.AsSpan()));
+                Assert.Equal(found, cert);
+                found = Assert.Single(coll.FindByThumbprint(hashAlgorithm, thumbprintHexUpper.AsSpan()));
+                Assert.Equal(found, cert);
+            }
+        }
+
+        [Fact]
+        public static void FindByThumbprint_ByAlgorithm_ArgValidation_HashAlgorithm()
+        {
+            const string Thumbprint = "000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f";
+            X509Certificate2Collection coll = [];
+            AssertExtensions.Throws<ArgumentNullException>(
+                "hashAlgorithm",
+                () => coll.FindByThumbprint(default, Thumbprint));
+
+            AssertExtensions.Throws<ArgumentException>(
+                "hashAlgorithm",
+                () => coll.FindByThumbprint(new HashAlgorithmName(""), Thumbprint));
+        }
+
+        [Fact]
+        public static void FindByThumbprint_ByAlgorithm_ArgValidation_ThumbprintHex()
+        {
+            X509Certificate2Collection coll = [];
+            AssertExtensions.Throws<ArgumentNullException>(
+                "thumbprintHex",
+                () => coll.FindByThumbprint(HashAlgorithmName.SHA256, (string)null));
+        }
+
+        [Fact]
+        public static void FindByThumbprint_ByAlgorithm_ArgValidation_ThumbprintHexInvalidLength()
+        {
+            HashAlgorithmName hashAlgorithm = HashAlgorithmName.SHA256;
+
+            using (X509Certificate2 cert = new(TestData.CertWithTemplateData))
+            using (X509Certificate2 unrelated = new(TestData.CertWithPolicies))
+            {
+                byte[] thumbprint = CryptographicOperations.HashData(hashAlgorithm, cert.RawDataMemory.Span);
+                string thumbprintHexOdd = Convert.ToHexString(thumbprint)[..^1];
+                X509Certificate2Collection coll = [cert, unrelated];
+
+                AssertExtensions.Throws<ArgumentException>(
+                    "thumbprintHex",
+                    () => coll.FindByThumbprint(hashAlgorithm, thumbprintHexOdd));
+
+                AssertExtensions.Throws<ArgumentException>(
+                    "thumbprintHex",
+                    () => coll.FindByThumbprint(hashAlgorithm, thumbprintHexOdd.AsSpan()));
+            }
+        }
+
+        [Fact]
+        public static void FindByThumbprint_ByAlgorithm_ArgValidation_ThumbprintIsNotHex()
+        {
+            HashAlgorithmName hashAlgorithm = HashAlgorithmName.SHA256;
+
+            using (X509Certificate2 cert = new(TestData.CertWithTemplateData))
+            {
+                X509Certificate2Collection coll = [cert];
+
+                AssertExtensions.Throws<ArgumentException>(
+                    "thumbprintHex",
+                    () => coll.FindByThumbprint(hashAlgorithm, "not a thumbprint"));
```

Comment:
I believe the hex parser we're using doesn't permit (skip) whitespace.  Worth a test?

Reply:
Yep. Sure.